server/server.go

```diff
@@ -270,7 +271,32 @@ func (s *Server) Shutdown(ctx context.Context) error {
 	// crio.service restart!!!
 	s.cleanupSandboxesOnShutdown(ctx)
 
-	return s.ContainerServer.Shutdown()
+	if err := s.ContainerServer.Shutdown(); err != nil {
+		return err
+	}
+
+	// first, make sure we sync all storage changes
+	syscall.Sync()
+
```

Comment:
it will be safer to use Syncfs in addition to sync so we have some error code:

```
	f, err := os.OpenFile(store.GraphRoot(), os.O_RDONLY, 0755)
	if err != nil {
		...
	}
	defer f.Close()
	
	if err = unix.Syncfs(int(f.Fd())); err != nil {
		...
	}
```

so we are sure the file system holding the graphroot is correctly synced.

The cost should be minimal after the full sync and we have an error if something goes wrong

Reply:
I went for Fsync as it seems to be more of what we want